nav2_tasks/include/nav2_tasks/bt_action_node.hpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_TASKS__BT_ACTION_NODE_HPP_
+#define NAV2_TASKS__BT_ACTION_NODE_HPP_
+
+#include <string>
+#include <memory>
+#include <condition_variable>
+#include <mutex>
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_tasks/task_client.hpp"
+#include "behavior_tree_core/action_node.h"
+#include "behavior_tree_core/bt_factory.h"
+
+namespace nav2_tasks
+{
+
+template<class CommandMsg, class ResultMsg>
+class BtActionNode : public BT::ActionNode
+{
+public:
+  explicit BtActionNode(const std::string & action_name)
+  : BT::ActionNode(action_name), task_client_(nullptr)
+  {
+  }
+
+  BtActionNode(const std::string & action_name, const BT::NodeParameters & params)
+  : BT::ActionNode(action_name, params), task_client_(nullptr)
+  {
+  }
+
+  BtActionNode() = delete;
+
+  virtual ~BtActionNode()
+  {
+  }
+
+  BT::NodeStatus tick() override
+  {
+    if (task_client_ == nullptr) {
```

Comment:
Because of how the BehaviorTree.CPP library builds its trees. First, the nodes are all built and then they are all assigned the blackboard. So, can't do any blackboard-related work in the constructor.